records/generator.go

```diff
@@ -120,28 +128,31 @@ func (rg *RecordGenerator) loadFromMaster(ip string, port string) (sj state.Stat
 	req, err := http.NewRequest("GET", u.String(), nil)
 	if err != nil {
 		logging.Error.Println(err)
+		return state.State{}, err
 	}
 
 	req.Header.Set("Content-Type", "application/json")
 
-	client := &http.Client{}
-	resp, err := client.Do(req)
+	resp, err := rg.httpClient.Do(req)
 	if err != nil {
 		logging.Error.Println(err)
+		return state.State{}, err
 	}
 
+	defer util.IgnoreError(resp.Body.Close)
```

Comment:
this is an interesting idea. we normally don't care about close errors. not sure why we need to be explicit here with IgnoreError()